src/app.module.ts

```diff
@@ -2,17 +2,40 @@
 // see: https://docs.nestjs.com/modules
 
 import { Elysia } from "elysia";
-import UsersService from "./users/users.service";
-import UsersController from "./users/users.controller";
+import { UsersController } from "./users/users.controller";
 import { db } from "@/database.providers";
 
 // the word 'setup' (instead of  e.g. 'bootstrap') is in correspondence with the official elysiajs docs
 // see: https://elysiajs.com/patterns/dependency-injection.html#dependency-injection
 
-export const setup = () => {
-    const usersService = UsersService(db);
-    const usersController = UsersController(usersService);
+const setupDatabase = new Elysia({ name: "setup" }).state("db", db);
```

Comment:
What is the point of naming plugins via `config.name`? Is it due to [Plugin deduplication](https://elysiajs.com/concept/plugin.html#plugin-deduplication)?

Reply:
Yes, the commonly used plugins should be initialized only once